test/e2e/operator.go

```diff
@@ -719,3 +721,55 @@ var _ = Describe("ARO Operator - Cloud Provider Config ConfigMap", func() {
 		}).WithContext(ctx).WithTimeout(DefaultEventuallyTimeout).Should(Succeed())
 	})
 })
+
+var _ = Describe("ARO Operator - Control Plane MachineSets", func() {
+	const (
+		cpmsEnabled = operator.CPMSEnabled
+	)
+
+	getCpmsOrNil := func(ctx context.Context, name string, options metav1.GetOptions) (*machinev1.ControlPlaneMachineSet, error) {
+		cpms, err := clients.MachineAPI.MachineV1().ControlPlaneMachineSets("openshift-machine-api").Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return nil, err
+		}
+		return cpms, nil
+	}
+
+	BeforeEach(func(ctx context.Context) {
+		if err := discovery.ServerSupportsVersion(clients.Kubernetes.Discovery(), machinev1.GroupVersion); err != nil {
+			Skip("Cluster does not support machinev1 API, CPMS controller not present")
+		}
+	})
+
+	It("should ensure CPMS is Inactive", func(ctx context.Context) {
+		By("checking whether CPMS is disabled in ARO operator config")
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		if instance.Spec.OperatorFlags.GetSimpleBoolean(cpmsEnabled) {
+			Skip("CloudProviderConfig Controller is not enabled, skipping test")
```

Comment:
ope, thanks for the catch

Reply:
Need to swap the "not" in the message too. We skip the test if CPMS is "enabled" (enabling CPMS disables the controller, since the purpose of the controller is to disable CPMS).

Reply:
I thought it was fine like this? We're not saying "CPMS is not enabled"; we're saying "CPMS _controller_ is not enabled".

Reply:
The content of your new commit works for me though, since it makes it doubly clear.

Reply:
> I thought it was fine like this? We're not saying "CPMS is not enabled"; we're saying "CPMS _controller_ is not enabled".

I intend to reuse this same controller for handling whatever fixups are necessary once we can enable CPMS in ARO, so this controller will branch its logic depending on the state of the flag but will always operate (a departure from our previous conventions of operator flag handling but one that's intended to improve the operational experience of setting the flag, I can share internal discussions if you're interested). 

We'd end up replacing this test at that point anyways though.